src/language-server/logger.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {createWriteStream, WriteStream} from 'fs';
+import {IConnection} from 'vscode-languageserver/lib/main';
+
+import Settings from './settings';
+
+export class Logger {
+  private logFile: WriteStream|undefined = undefined;
+  private settings: Settings|undefined;
+  private queue: string[] = [];
+  constructor(private readonly connection: IConnection) {
+  }
+
+  hookupSettings(settings: Settings) {
+    settings.changeStream.listen(({older, newer}) => {
+      if (older.logToFile !== newer.logToFile) {
+        if (this.logFile) {
+          this.logFile.end();
+          this.logFile = undefined;
+        }
+        if (newer.logToFile) {
+          this.logFile = createWriteStream(newer.logToFile, {flags: 'a'});
```

Comment:
Maybe I'm overthinking this, but should we handle the case where the `logToFile` path is wrong? e.g. not set `this.logFile` in order to avoid other errors down the line (when we log)

Reply:
+1, added some try / catches so that logging never crashes other stuff.